### ISSUE TITLE
Class vs Classes

### ISSUE THREAD

**Issue by fearless23** [css-modules/css-modules#174]
To every element in HTML, CSS rules are applied one by one.  In CSS we group CSS rules under a class name, thus reducing our CSS code and giving same set of rules to elements.

In BEM, SMACSS or any other style, say <button class= "button button-normal"> but here button has say 10 CSS rules and button-normal has 3, we applied two group of styles, and classes can be re-used on other elements. But in HTML it will be applied one-by-one on HTML Elements/DOM.

Q1: **Then why don`t we have single class for every element**? (and then in CSS, we combine group of rules)
say
<button class= "button-normal-f456341">
while in css 
.button-normal-f456341 { 
@extend button rules
- more css rules}

@extend: sass/scss or css-modules method

other css file contains groups of rules which we can call classes, but in HTML each element should have ONLY ONE CLASS.( except for interactive where we have to show/hide class or onclick events).

Since Web components is the future, so each element has to be a unique thing in itself with an ID and that same as using id for html elements and then extend class(which is just a group of css rules).

Q2: *_'CSS in JS' vs 'JS in CSS' ? (inline js or *_
Javascript has changed drastically from 2008 onwards, but now its time for CSS. Talk by Glen mentions CSS in JS, (great talk by the way and pinpoints pains of CSS ), but why not JS in CSS. Since, every change made so far to JS eco-system is in direction of giving wings to HTML i.e do more with HTML tags, custom tags, add some syntax in <div onclick="" > or <div []=""> or <div value={{ }}> etc., why not have CSS properties or rules do interactive stuff.

Then while in CSS, since custom-elements (web component) has a name, and we are using single class, its same as id which can be same as name of element or custom HTML Tag
button-normal-f456341{
/\* css styles/rules_/
/_ import/ extend classes/group of css rules, variables*/
inline-js}

leave main js for data manuplation and heavy animation, but little interaction should be done with CSS.

Future CSS for styles and interaction without JS.

Fantasy/DreamCode

button-normal-f456341{
/\* css styles/rules_/
/_ import/ extend classes/group of css rules, variables*/
_onClick: toggle{
some css rules or import class which is pre-defined set of rules in this file or another file}_
}

like backend and frontend now talk with Data.json and things like json stringify

The JS In CSS approach takes data, and make interaction logic possible in CSS only within a component.
like done in react or angular data-binding or standard css forms like username, password
say

button-normal-f456341{
/\* css styles/rules_/
/_ import/ extend classes/group of css rules, variables_/
/_ import/declare animation like keyframes etc. */
class: [button, blue-button]; // class array to add css classes
_onClick: 
  toggle {
     some css rules or import class which is pre-defined set of rules in this file or another file;
  };_
color: user-input
}

All in all a functional CSS in future covers styling, layout, interactivity( animation, sounds, effects, data) made into simple CSS properties without compiling.

Q3: **Importing and exporting should be handled automatically.** When we make classes, custom html tags all should be exported and should be available in any new file that is created without declaring import.

abc.html has custom tags or elements with id`s by default are accesible to all files within a folder (_any level of nesting should not obscure that, may be a file import-export.js lies in root folder and handles all files with any level of nesting._)
classes.css have custom classes/ group of css rules or parameters/variables which are just available for import.

style.css styles our HTML elements with each having unique ID, then add group of css rules, onclick events in CSS, data-binding.

HTML on rocks = adding simple syntax to <div {{}} [] etc. with help of JS and Framework
JS on rocks = making the language more abstract, better shorthands
CSS on rocks = JS on rocks make HTML on rocks with frameworks, then converting that to CSS properties or syntax.

This makes things easier, flexible, managable and more human way of developing front-end without much javascript.

**Comment by TheSisb** [comment]
tl;dr?

**Comment by tivac** [comment]
Or at least some code formatting. I can't really parse out what you're trying to say since none of the code examples are readable.
